### PR TITLE
Common code review

### PR DESCRIPTION
**Pull request by Jyotsna-Shetty**

### CODE REVIEW

**Comment by Jyotsna-Shetty** [comment]
Android Studio arranges all the imports in alphabetical order. Even after manually changing it, the previous order is restored. Hence in some files the order of imports is the same as before.

File: app/src/main/java/com/example/vehicleinfocheck/Element.java [hunk 1]
```diff
@@ -1,9 +1,10 @@
 package com.example.vehicleinfocheck;
+// An element is a field present in the website: an input field, a button, etc.
 
 import android.util.Log;
 
-// An element is a field present in the website: an input field, a button, etc.
 // This class defines all methods pertaining to elements
+// However, this class is not used in this project as the required methods did not work
```
Comment (Dcruise546) [inline app/src/main/java/com/example/vehicleinfocheck/Element.java]: Move this comment to first line of this file indicating this file is not part of this project!

File: app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java [hunk 2]
```diff
@@ -72,49 +73,52 @@ public void onClick(View v) {
...
                 dispatchTakePictureIntent();
-            } else {//toast message appear when both conditions are false
+            } else {
+                //toast message appear when both conditions are false
                 Toast.makeText(this, "Camera Permission is Required to Use camera.", Toast.LENGTH_SHORT).show();
             }
         }
     }
-    // onActivityResult method is used to display and save image
+    // onActivityResult method is used to display and save image as data of this app
```
Comment (Dcruise546) [inline app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java]: either provide a blank space after a comment or don't. But follow this style throughout the project!

File: app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java [hunk 3]
```diff
@@ -72,49 +73,52 @@ public void onClick(View v) {
...
+        //checking  permission  by comparing request code of camera with request code passed to onRequestPermissionResult
         if (requestCode == CAMERA_PERM_CODE) {
             //checking grantResults array is empty
-            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED){//if both the conditions are true then camera permission is given
+            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED){  //camera permission given
                 dispatchTakePictureIntent();
-            } else {//toast message appear when both conditions are false
+            } else {
+                //toast message appear when both conditions are false
                 Toast.makeText(this, "Camera Permission is Required to Use camera.", Toast.LENGTH_SHORT).show();
```
Comment (Dcruise546) [inline app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java]: Why so many Caps?

File: app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java [hunk 4]
```diff
@@ -72,49 +73,52 @@ public void onClick(View v) {
...
     protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         if (requestCode == CAMERA_REQUEST_CODE) {
-            //checking resultCode
             if (resultCode == Activity.RESULT_OK) {
-                File f = new File(currentPhotoPath);//creating new file f from the currentPhotoPath
-                selectedImage.setVisibility(View.VISIBLE);
-                selectedImage.setImageURI(Uri.fromFile(f));//set image to imageview using uri
-                Log.d("tag", "Absolute Url of Image is " + Uri.fromFile(f));//display absolute url of the file
+                File f = new File(currentPhotoPath);  //creating file f from currentPhotoPath
```
Comment (Dcruise546) [inline app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java]: f is not a good variable name

File: app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java [hunk 5]
```diff
@@ -124,55 +128,59 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
...
-        // Ensure that there is a camera activity to handle the intent , whether camera permission is given to the activity
+        // ensures camera is ready
         if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
             // Create the File where the photo should go
             File photoFile = null;
+            //createImageFile method within try catch block to prevent IOException
             try {
-                photoFile = createImageFile();//createImageFile can throw IOException so put inside try catch block.Creating photoFile assigning to Create image file which returns image
+                photoFile = createImageFile();  //this returns image
             } catch (IOException ex) {
```
Comment (Dcruise546) [inline app/src/main/java/com/example/vehicleinfocheck/ImageActivity.java:180]: Do not keep unhandled exception! Always throw an error message or handle the exception (if there is any other possibility to solve the exception)

File: app/src/main/java/com/example/vehicleinfocheck/WebActivity.java [hunk 6]
```diff
@@ -19,28 +18,28 @@
...
 
         // OnClickListener triggered when CopyButton is clicked
         CopyButton.setOnClickListener(v -> {
             // Creating an instance of ClipboardManager class
             ClipboardManager clipboard = (ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
-            ClipData clip = ClipData.newPlainText("simple text", vehicleNumber.getText());    // Creates a clip with value = vehicleNumber
-            clipboard.setPrimaryClip(clip);
-            if (!vehicleNumber.getText().toString().trim().equals("")) {
-                Toast.makeText(this, "Copied to clipboard", Toast.LENGTH_SHORT).show();// Displayed only if there is text to copy
+            if (!vehicleNumber.getText().toString().trim().equals("")) {                        // Clip is made only if there is text to copy
```
Comment (Dcruise546) [inline app/src/main/java/com/example/vehicleinfocheck/WebActivity.java]: Shouldn't there be a condition to check if the text is empty and throw an error message?

**Review by Dcruise546 — CHANGES_REQUESTED** [review]
Overall, the code is okay.
I felt the comments were a bit too much! But it's my personal opinion.

General questions:
Why there are multiple instances of 'png' files?

After making the requested changes, please directly merge them without requesting another review!

**Pull request merged**